### DISCUSSION TITLE
Handling Loading/Error states in Refine

### DISCUSSION THREAD

**Discussion by vinclou** [refinedev/refine#2506] [General]
```jsx
const { tableQueryResult, tableProps, searchFormProps } = useTable<Example, HttpError, ExampleParams>({
...someProps
})
```

Do you expect users of the framework to handle stuff like this:
```jsx
if (tableQueryResult.isLoading) {
  // do something
}

if (tableQueryResult.Error) {
  // do something
}
```

Or does Refine handle it under the hood?

**Comment by omeraplak** [comment]
Hey @vinclou 👋,
We handle the `loading` and `error` states in the packages provided by us. So you don't need to handle them. 
Here you can find all our table examples,
https://github.com/pankod/refine/tree/next/examples/table
